rosidl_generator_py/resource/_msg.py.em

```diff
@@ -290,10 +292,12 @@ if isinstance(type_, AbstractNestedType):
 @[end for]@
     )
 
-    def __init__(self, **kwargs):
-        assert all('_' + key in self.__slots__ for key in kwargs.keys()), \
-            'Invalid arguments passed to constructor: %s' % \
-            ', '.join(sorted(k for k in kwargs.keys() if '_' + k not in self.__slots__))
+    def __init__(self, check_fields = False, **kwargs):
+        self._check_fields = check_fields
```

Comment:
As part of the discussion of this PR I want to comment that my initial solution, besides having a new argument on the constructor of the messages, also retrieved a flag that could also be used to enable the assertions. The way I was doing it was basically the following:

```suggestion
    from os import getenv
    def __init__(self, check_fields = False, **kwargs):
        self._check_fields = check_fields or getenv("ROS_PYTHON_CHECK_FIELDS")
```
That way the member `self._check_fields` could be configured via the constructor of each message or just by setting the an environment variable to `True`.

I removed this solution because after testing it out, the time it took to retrieve the environment value basically made the solution's improvement almost non relevant. I believe that incorporating a way for the user to select to have the fields checked or not by just configuring a single variable for the whole node/project could be useful for the users. I would like to hear your thoughts on this @sloretz @clalancette